lib/block-supports/block-visibility.php

```diff
@@ -19,10 +44,80 @@ function gutenberg_render_block_visibility_support( $block_content, $block ) {
 		return $block_content;
 	}
 
-	if ( isset( $block['attrs']['metadata']['blockVisibility'] ) && false === $block['attrs']['metadata']['blockVisibility'] ) {
+	if ( ! isset( $block['attrs']['metadata']['blockVisibility'] ) ) {
+		return $block_content;
+	}
+
+	$block_visibility = $block['attrs']['metadata']['blockVisibility'];
+
+	// If blockVisibility is false, hide the block completely (original behavior).
+	if ( false === $block_visibility ) {
 		return '';
 	}
 
+	// If blockVisibility is an object, handle responsive visibility.
+	if ( is_array( $block_visibility ) ) {
+		$allowed_devices = array( 'desktop', 'tablet', 'mobile' );
+		$hidden_devices  = array();
+
+		foreach ( $block_visibility as $device => $is_hidden ) {
+			// Only allow whitelisted device names.
+			if ( false === $is_hidden && in_array( $device, $allowed_devices, true ) ) {
+				$hidden_devices[] = $device;
+			}
+		}
+
+		// If all devices are hidden, return empty.
+		if ( count( $hidden_devices ) === 3 &&
```

Comment:
Ideally, we don't hardcode this. Maybe at some point we allow adding additional (or intentionally removing) allowed devices. 

```suggestion
		if ( count( $hidden_devices ) === count( $allowed_devices ) &&
```

Reply:
Then we'd make that filterable and adapt. For now it's a good way to ensure no extraneous info sneaks in.